hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/extdataset/ExternalDatasetImpl.java

```diff
@@ -477,4 +482,7 @@ public MountVolumeMap getMountVolumeMap() {
   public List<FsVolumeImpl> getVolumeList() {
     return null;
   }
+
+  @Override
+  public void setLastDirScannerFinishTime(long time) {}
```

Comment:
Is there a reason that in one test case, we throw an unsupported exception while in this test class, it is a no-op? can we make both behave the same way?

Reply:
@xinglin , sir. I do not know why exacltly, I think the SimulatedFSDataset is just for simplifing test.  if we make both behave the same way, there is no need to have class SimulatedFSDataset.

Reply:
@xinglin Hi, sir. I do not know why exacltly, but i think the SimulatedFSDataset is just for simplifing test. if we make both behave the same way, there is no need to have class SimulatedFSDataset. What's your idea？Thanks a lot.

Reply:
+1, I am also confused why this behavior is different with ExternalDatasetImpl#setLastDirScannerFinishTime.